pipelines/edamam/__init__.py

```diff
@@ -0,0 +1,42 @@
+"""
+A pipeline for getting calories data via edamam API
+"""
+
+import datetime
+from typing import Any, Iterator, List, Sequence
+
+import dlt
+from dlt.common.typing import TDataItem
+from dlt.extract.source import DltResource
+from dlt.sources.helpers import requests
+
+EDAMAM_NUTRITION_DATA_API_URL = 'https://api.edamam.com/api/nutrition-data'
+current_time = datetime.datetime.now()
+
+@dlt.source(name='edamam')
+def edamam_source(api_secrets_key: str = dlt.secrets.value) -> Sequence[DltResource]:
```

Comment:
I think for such a simple pipeline you do not even need to declare the source. it is perfectly fine to just use standalone resource as below, but if you declare it, allow to pass the full credentials and also the ingredients. but as said: if you expose the resource as below you just need one.